elytron/WFLY-19314-oidc-logout-support.adoc

```diff
@@ -0,0 +1,221 @@
+- It can be a sequence of path segments (e.g. "/call/back/home")
+
+==== RP-Initiated Logout
+For RP-Initiated Logout, we'll introduce the following new configuration option:
+
+* `post-logout-redirect-uri`: This attribute must be an absolute URI.  It must be registered with the OpenId Provider.  This URI will be called after logout is executed.
+
+
+==== Front-Channel Logout and Back-Channel Logout
+
+For Front-Channel and Back-Channel Logout, we'll introduce the following new configuration options:
+
+* `logout-callback-path`: This is an optional property. It designates the path component of a URL
+which is called after logout has completed.  The declared URL path is appended to the client's application
+context URL.  The default path is `/logout/callback`.
+- The option can not be declared as an empty string or "/" only
+- It is expected that the text start with a forward slash (e.g. "/").
+- It can be a sequence of path segments (e.g. "/call/back/home")
+
+The value of this property can also be an absolute URL.  When the callback URL is outside the client
+application's context an absolute URL is required.
+
+For both scenarios the absolute URL must be registed with the OpenId Provider.
+
+```
+Example
+	The applications URL is https://localhost:8080/mySecureApp.  Using the default `logout-callback-path` value after logout the user would be sent to address of https://localhost:8080/mySecureApp/logout/callback .
+
+	The applications URL is https://localhost:8080/mySecureApp. If the callback address is to be https://www.wildfly.org/, the user would need to declare `logout-callback-path` to be `https://www.wildfly.org/`
+```
+
+
+* `logout-session-required`: A boolean value. The default value is true
+   - For Front-Channel Logout the boolean value specifies whether the request requires the iss (issuer) and sid (session ID) query parameters be included to identify the session with the OpenId Provider.
+   - For Back-Channel Logout the boolean value specifies whether the request requires the sid (session ID) Claim be included in the Logout Token to identify the session with the OpenId Provider.  This value must be `true` for back-channel logout to be executed.  If the value is `false` the `sid` will be marked for invalidation using a bounded map so that subsequent requests from the user will force the session to be invalidated.
+
+* `back-channel-logout-session-invalidation-limit`: The maximum size for a bounded map when marking sessions for invalidation during back-channel logout. Once this maximum size has been reached, the eldest entry will be removed.  The initial size is 16.
+
+=== Testing Configuration option values.
+Tests will be created to check the restrictions to the configuration option values.  An exception will be thrown when the value is invalid.
+
+* `logout-path`
+- The value can not be the empty string
+- The value can not be "/" only
+- It must start with the "/" character
+* `post-logout-uri`
```

Comment:
This should now be updated to reference the new name.

back-channel-logout-session-invalidation-limit should be mentioned here too.

It would also be good to mention doing some manual tests to verify that logout can be used with an OpenID provider other than Keycloak.